packages/@blockprotocol/graph/src/type-resolver.ts

```diff
@@ -0,0 +1,53 @@
+import { VersionedUri } from "@blockprotocol/type-system";
+import Ajv2020 from "ajv/dist/2020.js";
+import addFormats from "ajv-formats";
+
+import { EntityType } from "./type-resolver/entity-type-meta-schema.gen";
+import entityTypeMetaSchema from "./type-resolver/entity-type-meta-schema.json" assert { type: "json" };
+import { entityTypeToTypescript } from "./type-resolver/entity-type-to-typescript.js";
+import { fetchTypeAsJson } from "./type-resolver/shared.js";
+
+const ajv = new Ajv2020();
+addFormats(ajv);
+
+const entityTypeValidator = await ajv.compile<EntityType>(entityTypeMetaSchema);
+
+/**
+ * Validates that the schema at a given URI is a valid Entity Type
+ * @param versionedUri the URI / $id of the schema
+ * @throws if there are validation errors or if the schema is unreachable
+ */
+export const fetchAndValidateEntityType = async (
+  versionedUri: VersionedUri,
+) => {
+  const entityTypeSchema = await fetchTypeAsJson(versionedUri);
+
+  if (!entityTypeValidator(entityTypeSchema)) {
+    // eslint-disable-next-line no-console -- useful for debugging, intended as a CLI tool
+    console.error(
+      `Error validating entity type schema at ${versionedUri}:\n`,
+      JSON.stringify(entityTypeValidator.errors, undefined, 2),
+    );
+    throw entityTypeValidator.errors;
+  }
+
+  return entityTypeSchema;
+};
+
+/**
+ * Generates a string containing TypeScript type definitions for a given Entity Type
+ * @param versionedUri the URI at which the Entity Type is available
+ * @param depth how many links to follow when generating types.
+ *   - 0 will generate types for the Entity Type's properties only
+ *   - 1 will generate types for the links from the Entity Type and their possible destination entities
+ *   - 2 will generate types for the linked entities plus _their_ links and destinations, and so on
+ * @param outputDirectory
+ */
+export const generateTypeScriptFromEntityType = async (
```

Comment:
Agree with consistency, have gone for TypeScript in https://github.com/blockprotocol/blockprotocol/pull/861/commits/0f7c29dc7ba20f6d3ff202ca4bc26bb2a387c8e9

if it causes an issue with conversion libraries we can revisit, but in the absence of that happening that's how you spell it :P

Reply:
It's not how you spell it in camelCase though, as they aren't two separate words 👀 
If you wrote this in snake_case would you write `generate_type_script_from_entity_type`?

Definitely not going to block on this PR and if you prefer it this way I'm fine with it, but I've started really leaning the other way with this exact situation because of some of my work with integrations and such. (I had started to write "GitHub" for example in code and ended up getting generated stuff that looked like "Git Hub User" 🤮)

Reply:
I could be convinced the other way, I am probably anchored to `TypeScript` because my "correct a typo" instinct twitches when I see `Typescript` / `Javascritp`. To be continued another day when it crops up again